src/Serialization/main.jl

```diff
@@ -1,9 +1,135 @@
+################################################################################
+# Description of the saving and loading mechanisms
+#
+# We require that any types serialized through OSCAR are registered using the
+# @register_serialization_type macro, like so
+
+# @register_serialization_type NewType "String Representation of type" uses_id uses_params
+
+# The last three arguments are optional and can arise in any order. Passing a String
+# arguement will override how the type is store as a string. The last two are boolean
+# flags. When setting uses_id the object will be stored as a reference and will be
```

Comment:
```suggestion
# flags. When `uses_id` is given then instances of this type will be stored as a reference and will be
```
This explanation is unclear to me. Surely *at some point* the object needs to be stored. I think what is meant is that the object is only serialized once, and all places using it (e.g. if it occurs in a container) reference it indirectly.

And then I would also explain what happens *otherwise*. Namely the object is written immediately whenever it occurs. And if it is referenced in multiple places (e.g. it is twice in a vector), then we write its data twice (and so when loading, we end up with two objects).

Is that right?

If so, some alternative names for this flag come to mind:
- `by_ref` ("by reference" is how we access it)
- `unique` (as we preserve the property that this object is unique -- if serialized multiple times, then after deserialization in all places that referenced it, the same (identical) object is used
- `preserve_identity` (variation of the previous point)

Reply:
this is not exactly right. This actually has to do with not having a fixed normal form, essentially thats when it should be used. ids are not for elemen types they are for parent types. i.e. types that have encodings, for example some quotient ring will have a normal form that depends on the ideal. You can figure out when two elements are the same given the how they are encoded with respect to their parent. But I see your point I will add more about this in the next PR.